### PR TITLE
Noc benchmark design final 2

### PR DESCRIPTION
**Pull request by Srivat97**
noc benchmarks
#### Types of changes

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:

- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

### CODE REVIEW

File: vtr_flow/scripts/python_libs/vtr/task.py [hunk 1]
```diff
@@ -497,12 +497,24 @@ def create_job(
     Create an individual job with the specified parameters
     """
     param_string = "common" + (("_" + param.replace(" ", "_")) if param else "")
-    for spec_char in [":", "<", ">", "|", "*", "?"]:
+    
+    #remove any address-related characters that might be in the param_string
```
Comment (vaughnbetz) [inline vtr_flow/scripts/python_libs/vtr/task.py:501]: Could say what these address characters are and how you're replacing them @saaramahmoudi

File: vtr_flow/scripts/python_libs/vtr/parse_vtr_task.py [hunk 2]
```diff
@@ -209,6 +209,9 @@ def parse_task(config, config_jobs, flow_metrics_basename=FIRST_PARSE_FILE, alt_
     max_circuit_len = len("circuit")
     for job in config_jobs:
         work_dir = job.work_dir(get_latest_run_dir(find_task_dir(config, alt_tasks_dir)))
+        job.parse_command()[0] = work_dir
```
Comment (vaughnbetz) [inline vtr_flow/scripts/python_libs/vtr/parse_vtr_task.py:212]: Not sure if a comment would be helpful here or this is obvious in context @saaramahmoudi

File: vtr_flow/parse/parse_config/vpr_noc.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+%include "common/vtr_benchmarks.txt"
+
+%include "common/vpr.common.txt"
+%include "timing/vpr.pack.txt"
+%include "timing/vpr.place.txt"
+%include "timing/vpr.route_min_chan_width.txt"
+%include "timing/vpr.route_relaxed_chan_width.txt"
+
+NoC_agg_bandwidth;vpr.out;NoC Placement Costs. noc_aggregate_bandwidth_cost: (.*), noc_latency_cost: .*,
+NoC_latency;vpr.out;NoC Placement Costs. noc_aggregate_bandwidth_cost: .*, noc_latency_cost: (.*),
```
Comment (vaughnbetz) [inline vtr_flow/parse/parse_config/vpr_noc.txt:15]: Probably should also add a parse rule for latency constraints met or not (or how many failed) and some test for that case (constraints we can meet and check we do meet them).

**Comment by vaughnbetz** [comment]
Looks good; some suggestions for @saaramahmoudi which can be made after merge in the interests of time.

**Comment by vaughnbetz** [comment]
@saaramahmoudi @Srivat97 Nag, nag ... I think we just need to do a make format on the python part and merge this.  Sara or Srivatsan, could you do that? The commenting changes can go in later if they're going to hold things up.

We want to get this in so artifact evaluation can start.

**Comment by vaughnbetz** [comment]
There are some CI failures, but they are:
- QoR changes on a few small designs; these are due to the change to Ubuntu 22.04 for CI. @tangxifan is checking in new data for those, but one is a NoC design in vtr_reg_strong (complex_2_noc_1d_chain) so @Srivat97 should check in new QoR data for that one.
- Code formatting is broken in Ubuntu 22.04 for now; @tangxifan is also fixing that.
- Hence I'm going to merge this even though CI isn't green, as we've confirmed all issues are outside this actual change.

**Comment by vaughnbetz** [comment]
Not sure if the last run used Ubuntu 22.04 or not; just updated to the latest branch which kicked off CI again.

**Comment by Srivat97** [comment]
Confirmed with sara that the "complex_2_noc_1d_chain" benchmark QoR failure does not occur locally. There were a few more QoR failures related to NoC benchmarks (found in vtr_reg_nightly_test5) but they only on the pull_request run but passed during the push run (seems like not an actual issue as it was previously passing as well). 

Can we merge this then without making any more changes after the remaining tests finish?

**Comment by vaughnbetz** [comment]
A few QoR failures (seem random), and an incremental timing analysis mismatch it appears.  Also some clang versions that aren't working yet (or ever, and will be deprecated) in CI. These all look unrelated to your code. Merging.

**Pull request merged**